src/o365/spfx/commands/project/project-upgrade/upgrade-1.8.2.ts

```diff
@@ -38,7 +38,7 @@ module.exports = [
   new FN002003_DEVDEP_microsoft_sp_webpart_workbench('1.8.2'),
   new FN002009_DEVDEP_microsoft_sp_tslint_rules('1.8.2'),
   new FN002010_DEVDEP_microsoft_rush_stack_compiler_2_7('', false),
-  new FN002011_DEVDEP_microsoft_rush_stack_compiler_2_9('0.7.7'),
+  new FN002011_DEVDEP_microsoft_rush_stack_compiler_2_9('0.7.7', false),
```

Comment:
2.8, 3.0, 3.1 3.2. 3.3. 3.4 3.5 3.6. 3.7?

Reply:
There are suppose to be added by the developer themselves. We can account for those, but it would be more as an enhancement in a separate task.

@waldekmastykarz, what do you think. Should we account for the rest of the Rush modules. It might get crowded there, the more of those Microsoft releases :)

Reply:
could we instead have an approach where the compiler version is passed as a parameter? This way we avoid having to implement a rule per compiler version.
Something like
```TypeScript
new FN002010_DEVDEP_microsoft_rush_stack_compiler('2.9', 0.7.7', false),
new FN002010_DEVDEP_microsoft_rush_stack_compiler('3.0', 0.7.7', false),
new FN002010_DEVDEP_microsoft_rush_stack_compiler('3.1', 0.7.7', false), // etc...
```

Reply:
While it's tempting, it would block us from defining superseded rules, eg. you have a 1.8.2 project and you want to upgrade it to 1.10 (theory). For v1.9 we recommend upgrading to stack 2.9 but for 1.10 you should use 3.1 instead. If it's one and the same rule, then we don't have any mechanism in place, other than custom logic to supersede one finding with another. If it's two separate rules, then we can easily define that 3.1 supersedes 2.9 and the CLI will automatically handle it when generating the report.